app/src/shared/hooks/usePreventNavigatingAway.ts

```diff
@@ -85,24 +92,64 @@ export default function usePreventNavigatingAway({
     )
 }
 
+function patchBeforeUnload() {
+    if (window.addEventListener !== Window.prototype.addEventListener) {
+        return
+    }
+
+    function addEventListener(type: any, ...rest: [any, any]) {
+        if (type === 'beforeunload') {
+            return
+        }
+
+        Window.prototype.addEventListener.call(window, type.replace(/^hub_/, ''), ...rest)
+    }
+
+    window.addEventListener = addEventListener
+
+    function removeEventListener(type: any, ...rest: [any, any]) {
+        if (type === 'beforeunload') {
+            return
+        }
+
+        Window.prototype.removeEventListener.call(
+            window,
+            type.replace(/^hub_/, ''),
+            ...rest,
+        )
+    }
+
+    window.removeEventListener = removeEventListener
+}
+
 export function useBlockHistoryEffect() {
     const { blockers } = useBlockerStore()
 
-    const history = useHistory()
+    const [cache, bump] = useReducer((x) => x + 1, 0)
 
+    /**
+     * On each change to `blockers` we reblock history using the new set
+     * of potential blockers.
+     */
     useEffect(() => {
-        /**
-         * On each change to `blockers` we reblock history using the new set
-         * of potential blockers.
-         */
-        return history.block(({ pathname }) => {
+        patchBeforeUnload()
+
+        const unblock = history.block(({ retry, location: { pathname } }) => {
             const blocker = Object.values(blockers).find((blocker) => {
                 return blocker?.isDirty(pathname)
             })
 
-            if (blocker) {
-                return blocker.message
+            if (!blocker || confirm(blocker.message)) {
+                unblock()
+
+                retry()
+
+                bump()
```

Comment:
Reblock after unblocking. We have to do it manually now because `unblock` drops further blocking. If you, for example, go from Overview page to Connect page (no blockers between) then w/o this bump you'd be able to navigate away even with dirty state.